src/auth/openid-manager.ts

```diff
@@ -0,0 +1,187 @@
+import * as client from 'openid-client';
+import PromiseQueue from '../util/promise-queue';
+import { LocalStorageSync } from '../util/localstorage-proxy';
+
+interface RequestState {
+  verifier: string;
+  challenge: string;
+  state: string;
+}
+
+export type Whoami = Record<string, any>;
+
+const OPENID_SCOPES = 'openid email profile offline_access';
+let openidManagerSingleton: OpenidManager | undefined;
+
+export function getOpenIdManager(): OpenidManager {
+  if (!openidManagerSingleton) {
+    openidManagerSingleton = new OpenidManager();
```

Comment:
this singleton is hella-important to fight the "mount/unmount/remount" pattern of react components in dev mode.

It's another reason to consider having a store, with a global lifetime unaffected by that